modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBDecision.java

```diff
@@ -52,8 +78,41 @@ public Expression decisionNot(){
      *          ["!=", null, null]: boolean
      * @return
      */
-    public Expression decisionNotEqual(){
-        return null;
+    private Expression decisionNotEqual() {
+        // validate the arg list
+        if (json.size() != 3) {
+            throwUnexpectedArgumentCount(NOT_EQUALS, 2);
+        }
+        // first arg needs to be the operator
+        final String notEqualsOp = parse.get(json, 0);
+        if (!NOT_EQUALS.equals(notEqualsOp)) {
+            // not a NOT_EQUALS(!=) op
+            throwUnexpectedOperationException(NOT_EQUALS, notEqualsOp);
+        }
+        // get the comparables, treat nulls as literals
+        Object comparable1 = parse.value(json, 1, true);
+        Object comparable2 = parse.value(json, 2, true);
+        // handle nulls first
+        if (comparable1 == null) {
+            // not equal if comp2 is not null
+            return ff.literal(comparable2 != null);
+        }
+        // comp1 is not null
+        if (comparable2 == null) {
+            // not equal
+            return ff.literal(true);
+        }
+        // filter functions seem to determine Booleans equal to their string representations,
+        // so we need to ensure types are equal first
+        final Class comparable1Class = comparable1.getClass();
+        final Class comparabel2Class = comparable2.getClass();
+        // if the class types aren't equal, by definition the values are not equal
+        if (!comparable1Class.equals(comparabel2Class)) {
+            // class types are not equal
+            return ff.literal(true);
+        }
+        // 2 non-null comparables of the same class type, return the function
+        return ff.function("notEqualTo", ff.literal(comparable1), ff.literal(comparable2));
```

Comment:
Again, comparable1 and comparable2 are not guaranteed to be literals - they may be expressions (e.g. other functions)